Mage/src/main/java/mage/game/GameImpl.java

```diff
@@ -2081,7 +2077,7 @@ public UUID addDelayedTriggeredAbility(DelayedTriggeredAbility delayedAbility, A
         DelayedTriggeredAbility newAbility = delayedAbility.copy();
         newAbility.newId();
         if (source != null) {
-            newAbility.setSourceObjectZoneChangeCounter(getState().getZoneChangeCounter(source.getSourceId()));
+            newAbility.updateSourceObjectZoneChangeCounter(this, true);
             newAbility.setSourcePermanentTransformCount(this);
```

Comment:
Looks like `setSourcePermanentTransformCount` miss from all other `setSourceObjectZoneChangeCounter` usages. It can be a bug -- need research. `setSourcePermanentTransformCount` introduced with day/bound so dev might have forgotten to add it in other places.